### ISSUE TITLE
Sounds overlap between test set and train set

### ISSUE THREAD

**Issue by amiryanj** [ShanghaiTechCVDL/CIDNN#5]
> The train and set set has completely overlap

In file **create_dataset**.py, lines 118-120 the frame indices are shuffled and form the _train_idx_ and _test_idx_.
Let's assume frames 10 is in _test_idx_ and frame 9 is in _train_idx_.
Then when you call `get_samples_from_frame(f_i=9)`, it would use the information of frame 10 to generate the samples (lines 81:82).
Then let's say the test data almost perfectly exist in train_set as well!!!
Is there something I misunderstood?

**Comment by piaozhx** [comment]
Actually, you are right, It's an old version of the function `create_GC_train_test_data`, which is a mistake when we release `create_dataset.py`. 

an easy solution is: 
```
# wrong! it will cause overlap  between test set and train set
# random_idx = random.sample(range(frame_num), frame_num)

random_idx = range(frame_num)
```

**Comment by amiryanj** [comment]
Yes I fixed that mistake but it seems that results will converge somewhere else.
I run the algorithm for 350 epochs (10,000 is too much!) and the error does not come under 0.017.
![cidnn_result](https://user-images.githubusercontent.com/3169342/47144240-53bcb100-d2c7-11e8-9e1f-e3c9272e9a0c.png)
